apis/clusters/v1beta1/validation_test.go

```diff
@@ -0,0 +1,1072 @@
+package v1beta1
+
+import (
+	"testing"
+
+	"github.com/instaclustr/operator/pkg/models"
+)
+
+func TestCluster_ValidateCreation(t *testing.T) {
+	type fields struct {
+		Name                  string
+		Version               string
+		PCICompliance         bool
+		PrivateNetworkCluster bool
+		SLATier               string
+		TwoFactorDelete       []*TwoFactorDelete
+		Description           string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		wantErr bool
+	}{
+		{
+			name: "empty cluster name",
+			fields: fields{
+				Name: "",
+			},
+			wantErr: true,
+		},
+		{
+			name: "more than one two factor delete",
+			fields: fields{
+				Name: "test",
+				TwoFactorDelete: []*TwoFactorDelete{
+					{
+						Email: "test@mail.com",
+						Phone: "12345",
+					}, {
+						Email: "test@mail.com",
+						Phone: "12345",
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "unsupported SLAtier",
+			fields: fields{
+				Name:    "test",
+				SLATier: "test",
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid cluster",
+			fields: fields{
+				Name:    "test",
+				SLATier: "NON_PRODUCTION",
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Cluster{
+				Name:                  tt.fields.Name,
+				Version:               tt.fields.Version,
+				PCICompliance:         tt.fields.PCICompliance,
+				PrivateNetworkCluster: tt.fields.PrivateNetworkCluster,
+				SLATier:               tt.fields.SLATier,
+				TwoFactorDelete:       tt.fields.TwoFactorDelete,
+				Description:           tt.fields.Description,
+			}
+			if err := c.ValidateCreation(); (err != nil) != tt.wantErr {
+				t.Errorf("ValidateCreation() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestDataCentre_ValidateCreation(t *testing.T) {
+	type fields struct {
+		Name                  string
+		Region                string
+		CloudProvider         string
+		ProviderAccountName   string
+		CloudProviderSettings []*CloudProviderSettings
```

Comment:
There are no `CloudProviderSettings`, `NodeSize`, and `NodesNumber` fields inside the GenericDataCentreSpec. Could you clarify which specific aspects of the cluster specification validation are being tested in this pull request?

Reply:
The same case as we have above with `Cluster`